src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -125,45 +139,59 @@ internal SequencePosition Seek(in SequencePosition start, in SequencePosition en
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static SequencePosition SeekMultiSegment(object startSegment, int startIndex, object endSegment, int endPosition, long count, bool checkEndReachable)
+        private static SequencePosition SeekMultiSegment(object startSegment, int startIndex, object endSegment, int endPosition, long count)
         {
             var current = (ReadOnlySequenceSegment<T>)startSegment;
-            var end = (ReadOnlySequenceSegment<T>)endSegment;
 
-            SequencePosition result = default;
-            int currentIndex = startIndex;
+            int currentLength = current.Memory.Length - startIndex;
 
-            do
+            if (currentLength == count)
```

Comment:
This is not correct, we need to check if we can even go to Next, current might be the end of  ReadOnlySequence and we are not allowed to cross out.

Reply:
Interesting that none of the tests caught it.

Reply:
Looks like test gap. We should make sure our code coverage is higher: https://github.com/dotnet/corefx/issues/26768

Reply:
This code path was covered. Just not exact combination of things that might happen here.

Reply:
Might add some comments :)

Reply:
Was thinking there should be a fast path using `RunningIndex` rather than `Memory.Length` but I kept breaking the tests when I tried it

Reply:
It isn't possible which is why no test. It only goes down that path if `startObject != endObject` which for a single segment would mean `endObject` had to be null, which isn't allowed

Reply:
Added assert instead